shell/browser/native_window_views.cc

```diff
@@ -746,6 +746,24 @@ void NativeWindowViews::SetFullScreen(bool fullscreen) {
   // Note: the following must be after "widget()->SetFullscreen(fullscreen);"
   if (leaving_fullscreen && !IsVisible())
     FlipWindowStyle(GetAcceleratedWidget(), true, WS_VISIBLE);
+
+  // Auto-hide menubar when in fullscreen.
+  if (fullscreen) {
+    menu_bar_visible_before_fullscreen_ = IsMenuBarVisible();
+    SetMenuBarVisibility(false);
+  } else {
+    // No fullscreen -> fullscreen video -> un-fullscreen video results
```

Comment:
Is this only strictly true when fullscreening a video or is it anytime html fullscreen is requested? If a video isn't required, maybe we should update this to reflect that.

Reply:
I think it's the latter, that's why I called the test "correctly remembers state prior to HTML fullscreen transition". I just used the video example because that's one of the rare occasions where a HTML fullscreen transition can occur organically via user action instead of via JavaScript. I imagine the alternatives would illicit these reactions:

- "HTML fullscreen transition results...": "Don't know what that even is, probably doesn't matter"
- ".requestFullscreen() results...": "I as a developer never call that function, probably doesn't matter"
- "No fullscreen -> fullscreen video -> un-fullscreen video results": "Video elements are common, this covers an important edge case"

Reply:
Fair point on calling out the most understood use case.